### DISCUSSION TITLE
Backpropagation from the solution rather than the objective

### DISCUSSION THREAD

**Discussion by tisaac** [cvxpy/cvxpy#1453] [Q&A]
Hi folks,

As I understand it, `prob.backward()` gets derivatives of the objective with respect to the parameters of a problem.

But suppose for my work the objective of `prob` is incidental, and I only use it as a map from the parameters to the solution, which I further transform using some tools other than cvxopt.  That is, if `x* = argmin_x J(x,p)`, we can write `x* = S(p)`.  And if I have another quantity `q` and know `dq / dx*`, I would like to use backpropagation through `S` to get `dq / dp`.

Is there something like `prob.backward()` that does this?

**Comment by akshayka** [comment]
Yes! That would be PyTorch.

You can embed CVXPY into PyTorch using CVXPY Layers: https://github.com/cvxgrp/cvxpylayers
